### PR TITLE
Added update_marker() functionality to data_service

### PR DESCRIPTION
**Pull request by PCBush**
update_marker() will work if you get the original marker, change any attributes except for marker id, and then call update_marker with the modified LocationMarker object.

### CODE REVIEW

File: layers/shared_classes_layer/python/data_service.py [hunk 1]
```diff
@@ -69,3 +69,65 @@ def delete_marker(self, markerId):
...
+            original_marker_data = self.table.get_item(
+                Key={'markerId': str(marker_id)}
+            ).get('Item')
+
+            if not original_marker_data:
+                raise ValueError(f"Marker with ID {marker_id} does not exist")
+
+            #copy the original marker created date
+            original_date_created = original_marker_data['dateCreated']
+            marker.set_date_created(datetime.fromisoformat(original_date_created))
```
Comment (OrbitalDweller) [inline layers/shared_classes_layer/python/data_service.py]: @PCBush the copying of original date is redundant. Please remove it

File: layers/shared_classes_layer/python/data_service.py [hunk 2]
```diff
@@ -69,3 +69,65 @@ def delete_marker(self, markerId):
         except Exception as e:
             logger.error(f"Error deleting marker with markerId {markerId}: {e}")
             return None
+
+    def update_marker(self, marker: LocationMarker) -> LocationMarker:
```
Comment (OrbitalDweller) [inline layers/shared_classes_layer/python/data_service.py]: @PCBush no need to return LocationMarker here. Let's make it not return anything. This function raises an error, so this is enough to know if write was successful or not

File: layers/shared_classes_layer/python/data_service.py [hunk 3]
```diff
@@ -69,3 +69,65 @@ def delete_marker(self, markerId):
...
+        """
+        Update an existing marker in DynamoDB
+        :param marker: A LocationMarker with updated location details. Replaces marker with identical ID
```
Comment (OrbitalDweller) [inline layers/shared_classes_layer/python/data_service.py]: change `location details` to information. Not just location can be updated

File: layers/shared_classes_layer/python/data_service.py (hunk 1 above)
Comment (PCBush) [reply layers/shared_classes_layer/python/data_service.py]: Removed

File: layers/shared_classes_layer/python/data_service.py (hunk 2 above)
Comment (PCBush) [reply layers/shared_classes_layer/python/data_service.py]: Done

File: layers/shared_classes_layer/python/data_service.py (hunk 3 above)
Comment (PCBush) [reply layers/shared_classes_layer/python/data_service.py]: Corrected

**Review by OrbitalDweller — APPROVED** [review]

**Pull request merged**